src/cmd/linuxkit/pkgsrc/build.go

```diff
@@ -0,0 +1,142 @@
+package pkgsrc
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+)
+
+type buildOpts struct {
+	force bool
+	push  bool
+}
+
+// BuildOpt allows callers to specify options to Build
+type BuildOpt func(bo *buildOpts) error
+
+// WithBuildForce forces a build even if an image already exists
+func WithBuildForce() BuildOpt {
+	return func(bo *buildOpts) error {
+		bo.force = true
+		return nil
+	}
+}
+
+// WithBuildPush pushes the result of the build to the registry
+func WithBuildPush() BuildOpt {
+	return func(bo *buildOpts) error {
+		bo.push = true
+		return nil
+	}
+}
+
+// Build builds the package
+func (ps PkgSrc) Build(bos ...BuildOpt) error {
+	var bo buildOpts
+	for _, fn := range bos {
+		if err := fn(&bo); err != nil {
+			return err
+		}
+	}
+
+	if _, ok := os.LookupEnv("DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE"); !ok && ps.trust && bo.push {
+		return fmt.Errorf("Pushing with trust enabled requires $DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE to be set")
+	}
+
+	arch := runtime.GOARCH
+
+	if !ps.archSupported(arch) {
+		return fmt.Errorf("Arch %s not supported by this package", arch)
```

Comment:
erroring here is a a different to the current behaviour where we just print out a message but return with exit code 0. This will return (eventually) exit code 1 (if i read the code correctly).

I think printing a warning/info and exit with 0 is better in this case because we want to  just run `linuxkit pkg build` across all `./pkg` on all arches. If one of them errors because the package is not supported for the current arch then the rest of the build would fail.